File: qwencoder-eval/instruct/eval-dev-quality/testdata/ruby/transpile/pascalsTriangle/implementation/PascalsTriangle.java
package com.eval;

public class PascalsTriangle {
	int[][] generateTriangle(int rows) {
		if (rows < 0) {
			throw new IllegalArgumentException("Rows can't be negative!");
		}

		int[][] triangle = new int[rows][];

		for (int i = 0; i < rows; i++) {
			triangle[i] = new int[i + 1];
			triangle[i][0] = 1;
			for (int j = 1; j < i; j++) {
				triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
			}
			triangle[i][i] = 1;
		}
		return triangle;
	}
}

File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/transpile/pascalsTriangle/implementation/pascals_triangle.rb
def pascals_triangle(rows)
    if rows < 0
        raise ArgumentError, "Rows can't be negative!"
    end

    triangle = Array.new(rows)

    for i in 0...rows
        triangle[i] = Array.new(i + 1)
        triangle[i][0] = 1
        for j in 1...i
            triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j]
        end
        triangle[i][i] = 1
    end

    return triangle
end
